lib/OptimizationMOI/src/OptimizationMOI.jl

```diff
@@ -181,8 +188,22 @@ function MOI.eval_hessian_lagrangian(
     return
 end
 
-_create_new_optimizer(opt::MOI.AbstractOptimizer) = opt
-_create_new_optimizer(opt::MOI.OptimizerWithAttributes) = MOI.instantiate(opt)
+MOI.objective_expr(prob::MOIOptimizationProblem) = prob.f.expr
+MOI.constraint_expr(prob::MOIOptimizationProblem,i) = prob.f.cons_expr[i]
+
+function _create_new_optimizer(opt::MOI.OptimizerWithAttributes)
+    return _create_new_optimizer(MOI.instantiate(opt))
+end
+
+function _create_new_optimizer(model::MOI.AbstractOptimizer)
+    if MOI.supports_incremental_interface(model)
+        return model
+    end
+    return MOI.Utilities.CachingOptimizer(
+        MOI.Utilities.UniversalFallback(MOI.Utilities.Model{Float64}()),
+        model,
+    )
+end
```

Comment:
This enables support for AmplNLWriter.

Reply:
Thanks a lot for handling this, I had been unable to figure it out